src/multisig/sign/r1.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{
+    collections::TypedUsize,
+    crypto_tools::{k256_serde, rng},
+    gg20::sign::MessageDigest, // TODO put MessageDigest somewhere else
+    multisig::{self, keygen::SecretKeyShare},
+    sdk::{
+        api::{TofnFatal, TofnResult},
+        implementer_api::{serialize, RoundBuilder},
+    },
+};
+use ecdsa::{elliptic_curve::Field, hazmat::RecoverableSignPrimitive};
+use serde::{Deserialize, Serialize};
+
+use super::{r2, KeygenShareIds, SignProtocolBuilder, SignShareId};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Bcast {
+    pub(super) signature: k256_serde::Signature,
+}
+
+pub(super) fn start(
+    my_sign_id: TypedUsize<SignShareId>,
+    secret_key_share: SecretKeyShare,
+    msg_to_sign: &MessageDigest,
+    all_keygen_ids: KeygenShareIds,
+) -> TofnResult<SignProtocolBuilder> {
+    let msg_to_sign = k256::Scalar::from(msg_to_sign);
+    let signing_key = secret_key_share.share().signing_key().as_ref();
+
+    // TODO how best to sample ephemeral_scalar?
+    let rng = rng::rng_seed_sign(multisig::SIGN_TAG, my_sign_id, signing_key, &msg_to_sign)?;
```

Comment:
Hmm, lots of head scratching. So, traditional ECDSA generates `k` ephemerally, but this is considered a downside since bad RNG bits can break the scheme. Instead, we want to generate it deterministically from a message and the private key?

Looked into `k256` more. Looks like they also [generate k deterministically](https://docs.rs/k256/0.9.6/src/k256/ecdsa/sign.rs.html#114) for signing, using an [implementation for an RFC](https://docs.rs/ecdsa/0.12.4/ecdsa/rfc6979/fn.generate_k.html).

We can't use that method too since we don't have the digest state. What you have here looks close enough to me.

Reply:
I wonder if we should also do this kind of deterministic generation of `k` (for each share) in TSS. Is that why you added a shared `rng_seed_sign` method?

Reply:
beat you to it 😛 https://github.com/axelarnetwork/tofn/pull/185/commits/bc0a922dea371c00973d6e6d7746a78ac5716884

Makes sense to do this in GG20, too.  Currently we just use the OSs rng:
https://github.com/axelarnetwork/tofn/blob/6edd85215cc0ab0f881c92b1381cd85360b7dbd1/src/gg20/sign/r1.rs#L53-L54